src/MDFTesting.swift

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2019-present Google Inc.. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import XCTest
+
+extension XCTestCase {
+
+  public func MDFAssertEqualColors(
+    _ color1: UIColor,
+    _ color2: UIColor,
+    file: StaticString = #file,
+    line: UInt = #line
+  ) {
+    var red1: CGFloat
+    var green1: CGFloat
+    var blue1: CGFloat
+    var alpha1: CGFloat
+    (red1, green1, blue1, alpha1) = (0.0, 0.0, 0.0, 0.0)
+    color1.getRed(&red1, green: &green1, blue: &blue1, alpha: &alpha1)
+
+    var red2: CGFloat
+    var green2: CGFloat
+    var blue2: CGFloat
+    var alpha2: CGFloat
+    (red2, green2, blue2, alpha2) = (0.0, 0.0, 0.0, 0.0)
+    color2.getRed(&red2, green: &green2, blue: &blue2, alpha: &alpha2)
```

Comment:
did you try this with colors what have 2 components?

I have in the past had to check the number
if(CGColorGetNumberOfComponents(color.CGColor) == 2) {
sourced from: https://stackoverflow.com/questions/1496716/how-do-i-get-red-green-blue-rgb-and-alpha-back-from-a-uicolor-object

I believe grey scale colors have 2 components: Black, White, Grey

Reply:
the return value of the the getRed..blue... call tells you if it worked or not.

https://developer.apple.com/documentation/uikit/uicolor/1621919-getred?language=objc

Reply:
I think that is another way to compare the values. getRGB will return the RGB values. If it is a grey scale value, then RGB values should be the same.